src/appengine/handlers/cron/fuzz_strategy_selection.py

```diff
@@ -124,38 +113,60 @@
       GROUP BY
         strategy)))
 """
-BANDIT_PROBABILITY_QUERY = """
-SELECT
-  a.strategy AS strategy,
-  bandit_weight_high_temperature,
-  bandit_weight_low_temperature,
-  bandit_weight_medium_temperature,
-  a.run_count + b.run_count + c.run_count as run_count 
-FROM {high_temperature_query} a
-JOIN {low_temperature_query} b ON a.strategy = b.strategy
-JOIN {medium_temperature_query} c ON a.strategy = c.strategy
+
+STRATEGY_SUBQUERY_FORMAT = """
+IF
+  (strategy_{strategy_name} > 0,
+    "{strategy_name},",
+    "") AS strategy_{strategy_name},
 """
 
 
-def _query_multi_armed_bandit_probabilities():
+def _query_multi_armed_bandit_probabilities(engine_name, strategy_list):
   """Get query results.
 
   Queries above BANDIT_PROBABILITY_QUERY and yields results
   from bigquery. This query is sorted by strategies implemented."""
+  strategy_names_list = [
+      strategy_entry.name for strategy_entry in strategy_list
+  ]
+  strategies_subquery = '\n'.join([
+      STRATEGY_SUBQUERY_FORMAT.format(strategy_name=strategy_name)
+      for strategy_name in strategy_names_list
+  ])
   client = big_query.Client()
-  formatted_query = BANDIT_PROBABILITY_QUERY.format(
-      high_temperature_query=BANDIT_PROBABILITY_SUBQUERY.format(
+  formatted_query = BANDIT_PROBABILITY_QUERY_FORMAT.format(
+      high_temperature_query=BANDIT_PROBABILITY_SUBQUERY_FORMAT.format(
           temperature_type='high',
-          temperature_value=HIGH_TEMPERATURE_PARAMETER),
-      low_temperature_query=BANDIT_PROBABILITY_SUBQUERY.format(
-          temperature_type='low', temperature_value=LOW_TEMPERATURE_PARAMETER),
-      medium_temperature_query=BANDIT_PROBABILITY_SUBQUERY.format(
+          temperature_value=HIGH_TEMPERATURE_PARAMETER,
+          strategies=','.join([
```

Comment:
lines 142 - 145 are repeated three times, can you please generate the strategies string just once and then use it for all the queries?